drivers/storage/portworx/component/telemetry.go

```diff
@@ -1000,6 +1003,44 @@ func getArcusRegisterProxyURL(cluster *corev1.StorageCluster) string {
 	return productionArcusRegisterProxyURL
 }
 
+// CanAccessArcusRegisterEndpoint checks if telemetry registration endpoint is reachable
+func CanAccessArcusRegisterEndpoint(cluster *corev1.StorageCluster) (bool, error) {
+	endpoint := getArcusRegisterProxyURL(cluster)
+	logrus.Debugf("checking whether telemetry registration endpoint %s is accessible on cluster %s",
+		endpoint, cluster.Name)
+
+	url, err := url.Parse(fmt.Sprintf("https://%s:443/auth/1.0/ping", endpoint))
+	if err != nil {
+		return false, err
+	}
+
+	request := &http.Request{
+		Method: "GET",
+		URL:    url,
+		Header: map[string][]string{
+			"product-name": {"portworx"},
+			"appliance-id": {cluster.Status.ClusterUID},
+			"component-sn": {cluster.Name},
+		},
+	}
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		logrus.WithError(err).Warnf("failed to access telemetry registration endpoint %s", endpoint)
+		return false, nil
+	} else if response.StatusCode != 200 {
```

Comment:
we should still enable telemetry as long as the network access passes. i am thinking about several scenarios this will not return success:
1. the version is changed from 1.0 to 2.0 in the url
2. internal server error
3. the format of the request header/body changes